regression_detector.py

```diff
@@ -188,30 +215,41 @@ def process_regressions_into_gh_issue(regressions_dict, owner: str, output_path:
         output_path = args.output if args.output else get_default_output_path(control["name"])
         regressions_dict = generate_regression_dict(control, treatment)
         process_regressions_into_yaml(regressions_dict, output_path, args.control, args.treatment)
-    elif not args.control and args.treatment:
-        if not args.platform:
-            raise ValueError("A platform must be specified with the --platform flag to retrieve the "
-                             "previous metrics JSONs as control from S3.")
-        # Download control from S3
+        exit(0)
+
+    # Query S3 to get control and treatment json files
+    if not args.platform:
+        raise ValueError("A platform must be specified with the --platform flag to retrieve the "
+                         "previous metrics JSONs as control from S3.")
+    # User only provide the treatement file, and expect us to download from S3
+    control, treatment = None, None
+    if not args.control and args.treatment:
         json_path = Path(args.treatment)
         assert json_path.exists(), f"Specified result json path {args.treatment} does not exist."
-        date: str = get_date_from_metrics(json_path.stem)
+        end_date: datetime = datetime.strptime(get_date_from_metrics(json_path.stem), "%Y-%m-%d")
         userbenchmark_name: str = get_ub_name(args.treatment)
-        latest_metrics_jsons = get_latest_n_jsons_from_s3(1, userbenchmark_name, args.platform, date)
-
-        if len(latest_metrics_jsons) == 0:
-            raise RuntimeWarning("No previous JSONS found to compare against. No regression info has been generated.")
-
-        s3 = S3Client(USERBENCHMARK_S3_BUCKET, USERBENCHMARK_S3_OBJECT)
-        control = s3.get_file_as_json(latest_metrics_jsons[0])
         with open(json_path, "r") as cfptr:
             treatment = json.load(cfptr)
-        regressions_dict = generate_regression_dict(control, treatment)
-        output_path = args.output if args.output else get_default_output_path(control["name"])
-        process_regressions_into_yaml(regressions_dict, output_path, args.control, args.treatment)
-
-        owner = " ".join(args.owner) if args.owner else "@xuzhao9"
-        process_regressions_into_gh_issue(regressions_dict, owner, args.gh_issue_path, args.errors_path)
     else:
-        # S3 path
-        print("Comparison for metrics from Amazon S3 given a userbenchmark name is WIP.")
+        assert args.name, f"To detect regression with S3, you must specify a userbenchmark name."
+        end_date = datetime.strptime(args.end_date, "%Y-%m-%d")
+        userbenchmark_name = args.name
```

Comment:
more intuitive to switch these lines given the assertion above
```suggestion
        userbenchmark_name = args.name
        end_date = datetime.strptime(args.end_date, "%Y-%m-%d")
```